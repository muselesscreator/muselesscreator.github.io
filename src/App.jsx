import React from 'react';
import logo from './logo.svg';
import './App.scss';
import Sidebar from './components/Sidebar';
import Navbar from './components/Navbar';
import PageArticle from './components/PageArticle';
import { pages } from './data/pages';


function App() {
  const [activePage, setActivePage] = React.useState(pages.about);
  return (
    <main>
      <Sidebar />
      <div className="main-content">
        <Navbar  activePage={activePage} setActivePage={setActivePage} />
        <PageArticle activePage={activePage} />
      </div>
    </main>
  );
}

export default App;
