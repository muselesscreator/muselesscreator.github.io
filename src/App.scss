.mc-select-root {
  min-width: 150px;
}
label.mc-select-label {
  background-color: rgba(0, 0, 0, 0.5);
  color: #fee715 !important;
}
.MuiSvgIcon-root {
  color: #fee715 !important;
}
.MuiFormControl-root.mc-select-root {
  background-color: #101820 !important;
  border: 1px solid #fee715 !important;
  margin-left: 1rem;
  margin-right: 1rem;
  margin-top: .5rem;
  margin-bottom: .5rem;
  min-width: 100px;
  .mc-select {
    div[role="combobox"] {
      padding-top: 0px;
      padding-bottom: 0px;
      color: white;
      min-width: 50px;
    }
  }
}
.mc-select-item {
  background-color: #1e1e1e !important;
  color: white !important;
}
.MuiPaper-root {
  background-color: #101820 !important;
  border: 1px solid #fee715 !important;
}
