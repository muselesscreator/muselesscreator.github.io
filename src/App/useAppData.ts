import React from 'react';
import { useLocation } from 'react-router-dom';

import { pages } from '~/data/pages';


const useAppData = (): { activePage: string } => {
  const loadedPath = useLocation().pathname.slice(1);

  const effectivePage = loadedPath === '' ? pages.about : pages[loadedPath];

  const [activePage, setActivePage] = React.useState(
    loadedPath  !== '' ? pages[loadedPath] : pages.about);

  React.useEffect(() => {
    setActivePage(effectivePage);
  }, [effectivePage, setActivePage]);

  return { activePage };
}

export default useAppData;
