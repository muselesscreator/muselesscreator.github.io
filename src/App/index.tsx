import { Routes, Route } from 'react-router-dom';


import Sidebar from '~/components/Sidebar';
import Navbar from '~/components/Navbar';
import PageArticle from '~/components/PageArticle';
import { pages } from '~/data/pages';

import useAppData from './useAppData';
import './App.scss';


export const routes = [
  { path: '/', page: pages.about },
  { path: '/about', page: pages.about },
  { path: '/resume', page: pages.resume },
  { path: '/portfolio', page: pages.portfolio },
  { path: '/contact', page: pages.contact },
  { path: '/demo', page: pages.demo },
  { path: '/blog', page: pages.blog },
];

export const App = () => {
  const { activePage } = useAppData();
  return (
    <main>
      <Sidebar />
      <div className="main-content">
        <Navbar activePage={activePage} />
        <Routes>
          {routes.map(({ path, page }) => (
            <Route key={path} path={path} element={<PageArticle activePage={page} />} />
          ))}
        </Routes>
      </div>
    </main>
  );
};

export default App;
