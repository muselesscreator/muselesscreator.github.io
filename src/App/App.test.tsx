import {
  vi,
  describe,
  it,
  test,
  expect,
  Mock,
  beforeEach,
} from 'vitest';
import { shallow, ExplorerData } from '@muselesscreator/react-shallow-snapshot';

import useAppData from './useAppData';
import App, { routes } from '.';

vi.mock('react-router-dom', () => ({ Routes: 'Routes', Route: 'Route' }));
vi.mock('~/components/Sidebar', () => ({ default: 'Sidebar' }));
vi.mock('~/components/Navbar', () => ({ default: 'Navbar' }));
vi.mock('~/components/PageArticle', () => ({ default: 'PageArticle' }));
vi.mock('./useAppData', () => ({ default: vi.fn() }));

const hookProps = { activePage: 'test-page' };
(useAppData as Mock).mockReturnValue(hookProps);

let el: typeof ExplorerData;
describe('App', () => { 
  beforeEach(() => {
    el = shallow(<App />);
  });
  describe('behavior', () => {
    it('loads activePage from hooks', () => {
      expect(useAppData).toHaveBeenCalledWith();
    });
  });
  describe('output', () => {
    test('snapshot', () => {
      expect(el.snapshot).toMatchSnapshot();
    });
    it('renders a Sidebar component', () => {
      expect(el.instance.findByType('Sidebar')).toHaveLength(1);
    });
    it('renders Navbar with active page', () => {
      expect(el.instance.findByType('Navbar')[0].props.activePage).toBe(hookProps.activePage);
    });
    it('renders a Route for each route', () => {
      const routeEls = el.instance.findByType('Route');
      expect(routeEls).toHaveLength(routes.length);
      routeEls.forEach((_: typeof ExplorerData, i: number) => {
        const { props } = routeEls[i];
        const element = props.element;
        expect(props.path).toBe(routes[i].path);
        expect(element.type).toBe('PageArticle');
        expect(element.props.activePage).toBe(routes[i].page);
      });
    });
  });
});

