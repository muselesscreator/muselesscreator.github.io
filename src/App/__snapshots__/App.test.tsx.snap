// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > output > snapshot 1`] = `
<main>
  <Sidebar />
  <div
    className="main-content"
  >
    <Navbar
      activePage="test-page"
    />
    <Routes>
      <Route
        element={
          <PageArticle
            activePage="about"
          />
        }
        path="/"
      />
      <Route
        element={
          <PageArticle
            activePage="about"
          />
        }
        path="/about"
      />
      <Route
        element={
          <PageArticle
            activePage="resume"
          />
        }
        path="/resume"
      />
      <Route
        element={
          <PageArticle
            activePage="portfolio"
          />
        }
        path="/portfolio"
      />
      <Route
        element={
          <PageArticle
            activePage="contact"
          />
        }
        path="/contact"
      />
      <Route
        element={
          <PageArticle
            activePage="demo"
          />
        }
        path="/demo"
      />
      <Route
        element={
          <PageArticle
            activePage="blog"
          />
        }
        path="/blog"
      />
      <Route
        element={
          <PageArticle
            activePage="api"
          />
        }
        path="/api"
      />
    </Routes>
  </div>
</main>
`;
